src/test/java/uk/gov/justice/digital/delius/service/CustodyServiceTest.java

```diff
@@ -549,4 +554,164 @@ public void willCreateTelemetryEventStatingNewOneHasBeenInserted() {
             }
         }
     }
+
+    @Nested
+    @DisplayName("when calling getCustodyByBookNumber")
+    class WhenRetrievingByBookingNumber {
+        @Nested
+        @DisplayName("and there is more than one event for the booking number")
+        class WhenHasMultipleActiveEvents {
+            @BeforeEach
+            void setup() throws ConvictionService.DuplicateActiveCustodialConvictionsException {
+                when(offenderRepository.findByNomsNumber(anyString())).thenReturn(Optional.of(Offender.builder().offenderId(99L).build()));
+                when(convictionService.getSingleActiveConvictionByOffenderIdAndPrisonBookingNumber(anyLong(), anyString()))
+                        .thenThrow(new ConvictionService.DuplicateActiveCustodialConvictionsException(2));
+            }
+
+            @Test
+            @DisplayName("then a NotFoundException will be thrown")
```

Comment:
I really like this nesting of display names to get nice descriptions of the tests.  It's a real shame that neither Intellij (ignoring @DisplayName) and Gradle (only shows last 2 levels of nesting) are not handling this very well.